ci/docker/runtime_functions.sh

```diff
@@ -31,14 +31,49 @@ clean_repo() {
     git submodule update --init --recursive
 }
 
+# wrap compiler calls with ccache
+build_ccache_wrappers() {
+    set -ex
+
+    rm -f cc
+    rm -f cxx
+
+    touch cc
+    touch cxx
+
+    if [ -z ${CC+x} ]; then
+        echo "No \$CC set, defaulting to gcc";
+        export CC=gcc
+    fi
+
+    if [ -z ${CXX+x} ]; then
+       echo "No \$CXX set, defaulting to g++";
+       export CXX=g++
+    fi
+
+    # this function is nessesary for cuda enabled make based builds, since nvcc needs just an executable for -ccbin
+
+    echo -e "#!/bin/sh\n/usr/local/bin/ccache ${CC} \"\$@\"\n" >> cc
```

Comment:
why use touch and >> instead of a simple >?

Reply:
Agree, I think we should use >   doesn't make sense to append in case the file already exist does it?

Reply:
Yes, this was the first thing I have tried, but I had problems and could not get it to work. It seemed that EOF was not what it should be.